### PR TITLE
Error fix in `ScanInfo` for PrairieView scans

### PR DESCRIPTION
**Pull request by kushalbakshi**
**PR HIGHLIGHTS:**
This PR specifically addresses ingestion of single-plane scans into the ScanInfo table using the PrairieView loader. The existing codebase assumes the number of planes acquired is always > 1. I've added an `if - elif` statement between lines `587-655`. 

- [x] The code was tested locally on one image and `ScanInfo` was successfully populated. 
- [x] The code was formatted using Black. 
- [x] CHANGELOG and version.py updated

**PR DETAILS:**
Current code in `scan.py` results in the following error when attempting to populate `ScanInfo` and this PR will catch and prevent this from occurring. 
```In [18]: scan.ScanInfo.populate()
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [18], in <cell line: 1>()
----> 1 scan.ScanInfo.populate()

File /opt/conda/lib/python3.9/site-packages/datajoint/autopopulate.py:230, in AutoPopulate.populate(self, suppress_errors, return_exception_objects, reserve_jobs, order, limit, max_calls, display_progress, processes, make_kwargs, *restrictions)
    226 if processes == 1:
    227     for key in (
    228         tqdm(keys, desc=self.__class__.__name__) if display_progress else keys
    229     ):
--> 230         error = self._populate1(key, jobs, **populate_kwargs)
    231         if error is not None:
    232             error_list.append(error)

File /opt/conda/lib/python3.9/site-packages/datajoint/autopopulate.py:281, in AutoPopulate._populate1(self, key, jobs, suppress_errors, return_exception_objects, make_kwargs)
    279 self.__class__._allow_insert = True
    280 try:
--> 281     make(dict(key), **(make_kwargs or {}))
    282 except (KeyboardInterrupt, SystemExit, Exception) as error:
    283     try:

File /opt/conda/lib/python3.9/site-packages/element_calcium_imaging/scan.py:583, in ScanInfo.make(self, key)
    562     pvscan_info = prairieviewreader.get_pv_metadata(scan_filepaths[0])
    563     self.insert1(
    564         dict(
    565             key,
   (...)
    580         )
    581     )
--> 583     self.Field.insert(
    584         dict(
    585             key,
    586             field_idx=plane_idx,
    587             px_height=pvscan_info["height_in_pixels"],
    588             px_width=pvscan_info["width_in_pixels"],
    589             um_height=pvscan_info["height_in_um"],
    590             um_width=pvscan_info["width_in_um"],
    591             field_x=pvscan_info["fieldX"],
    592             field_y=pvscan_info["fieldY"],
    593             field_z=pvscan_info["fieldZ"][plane_idx],
    594         )
    595         for plane_idx in range(pvscan_info["num_planes"])
    596     )
    597 else:
    598     raise NotImplementedError(
    599         f"Loading routine not implemented for {acq_software} "
    600         "acquisition software"
    601     )

File /opt/conda/lib/python3.9/site-packages/datajoint/table.py:409, in Table.insert(self, rows, replace, skip_duplicates, ignore_extra_fields, allow_direct_insert)
    406     return
    408 field_list = []  # collects the field list from first row (passed by reference)
--> 409 rows = list(
    410     self.__make_row_to_insert(row, field_list, ignore_extra_fields)
    411     for row in rows
    412 )
    413 if rows:
    414     try:

File /opt/conda/lib/python3.9/site-packages/datajoint/table.py:409, in <genexpr>(.0)
    406     return
    408 field_list = []  # collects the field list from first row (passed by reference)
--> 409 rows = list(
    410     self.__make_row_to_insert(row, field_list, ignore_extra_fields)
    411     for row in rows
    412 )
    413 if rows:
    414     try:

File /opt/conda/lib/python3.9/site-packages/element_calcium_imaging/scan.py:593, in <genexpr>(.0)
    562     pvscan_info = prairieviewreader.get_pv_metadata(scan_filepaths[0])
    563     self.insert1(
    564         dict(
    565             key,
   (...)
    580         )
    581     )
    583     self.Field.insert(
    584         dict(
    585             key,
    586             field_idx=plane_idx,
    587             px_height=pvscan_info["height_in_pixels"],
    588             px_width=pvscan_info["width_in_pixels"],
    589             um_height=pvscan_info["height_in_um"],
    590             um_width=pvscan_info["width_in_um"],
    591             field_x=pvscan_info["fieldX"],
    592             field_y=pvscan_info["fieldY"],
--> 593             field_z=pvscan_info["fieldZ"][plane_idx],
    594         )
    595         for plane_idx in range(pvscan_info["num_planes"])
    596     )
    597 else:
    598     raise NotImplementedError(
    599         f"Loading routine not implemented for {acq_software} "
    600         "acquisition software"
    601     )

IndexError: invalid index to scalar variable.```

### CODE REVIEW

**Comment by tdincer** [comment]
@kushalbakshi The entries for n=1 and n>1 are created in an if condition. I would recommend creating the entry/entries at one block without any condition and ingest the entry/entries afterwards with a condition. This would half the line of codes.

File: element_calcium_imaging/scan.py [hunk 1]
```diff
@@ -606,18 +606,20 @@ def estimate_nd2_scan_duration(nd2_scan_obj):
...
-                        px_height=pvscan_info["height_in_pixels"],
-                        px_width=pvscan_info["width_in_pixels"],
-                        um_height=pvscan_info["height_in_um"],
-                        um_width=pvscan_info["width_in_um"],
-                        field_x=pvscan_info["fieldX"],
-                        field_y=pvscan_info["fieldY"],
-                        field_z=pvscan_info["fieldZ"][plane_index],
-                    )
-                    for plane_index in range(pvscan_info["num_planes"])
+                    [
```
Comment (ttngu207) [inline element_calcium_imaging/scan.py:609]: You don't need to turn this into a list, datajoint `.insert()` can handle a generator fine

File: element_calcium_imaging/scan.py [hunk 2]
```diff
@@ -641,17 +643,19 @@ def estimate_nd2_scan_duration(nd2_scan_obj):
...
-                        field_idx=0,
-                        px_height=pvscan_info["height_in_pixels"],
-                        px_width=pvscan_info["width_in_pixels"],
-                        um_height=pvscan_info["height_in_um"],
-                        um_width=pvscan_info["width_in_um"],
-                        field_x=pvscan_info["fieldX"],
-                        field_y=pvscan_info["fieldY"],
-                        field_z=pvscan_info["fieldZ"],
-                    )
+                    [
```
Comment (ttngu207) [inline element_calcium_imaging/scan.py:646]: instead of turning this into a list of a single dictionary, use `.insert1()`

Comment (kushalbakshi) [reply element_calcium_imaging/scan.py]: I didn't have it in this format originally but noticed the ingestion for the other acquisition softwares (ex. lines 463-482) use this format for the Field part table

**Comment by kushalbakshi** [comment]
@ttngu207 @tdincer, I've edited the code to be quite concise now. The if condition only exists within the Field table. This was tested on both single plane and multi plane metadata files and both populated `ScanInfo` and `Field` tables successfully.

File: element_calcium_imaging/scan.py [hunk 3]
```diff
@@ -581,18 +605,32 @@ def estimate_nd2_scan_duration(nd2_scan_obj):
...
+                        px_height=pvscan_info["height_in_pixels"],
+                        px_width=pvscan_info["width_in_pixels"],
+                        um_height=pvscan_info["height_in_um"],
+                        um_width=pvscan_info["width_in_um"],
+                        field_x=pvscan_info["fieldX"],
+                        field_y=pvscan_info["fieldY"],
+                        field_z=pvscan_info["fieldZ"][plane_idx],
+                    )
+                    for plane_idx in range(pvscan_info["num_planes"])
+                ]
```
Comment (ttngu207) [inline element_calcium_imaging/scan.py:633]:
```suggestion
                    dict(
                        key,
                        field_idx=0,
                        px_height=pvscan_info["height_in_pixels"],
                        px_width=pvscan_info["width_in_pixels"],
                        um_height=pvscan_info["height_in_um"],
                        um_width=pvscan_info["width_in_um"],
                        field_x=pvscan_info["fieldX"],
                        field_y=pvscan_info["fieldY"],
                        field_z=pvscan_info["fieldZ"][plane_idx] if pvscan_info["num_planes"] == 1 else pvscan_info["fieldZ"],
                    )
                    for plane_idx in range(pvscan_info["num_planes"])
```

Would this work?

Comment (tdincer) [reply element_calcium_imaging/scan.py:633]: I was thinking about this! If `pvscan_info["num_planes"]` doesn't exist for `num_planes>1`, we can condition the statment in `range`.

Comment (ttngu207) [reply element_calcium_imaging/scan.py:633]: I'd expect `pvscan_info["num_planes"]` to always exist

**Comment by kushalbakshi** [comment]
@ttngu207, I tested your suggestion with single and multi-plane and it works. Suggestion committed!

**Review by ttngu207 — APPROVED** [review]

**Comment by kabilar** [comment]
@kushalbakshi Please push a tag for this merged pull request (note that there have been commits to `main` since this pull request was merged) and add the following to the CHANGELOG:

```
[0.5.2]: https://github.com/datajoint/element-calcium-imaging/releases/tag/0.5.2
```

**Pull request merged**